src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -57,6 +56,53 @@ public WebScriptHostManager(ScriptHostConfiguration config) : base(config)
             return response;
         }
 
+        private static MethodInfo CreateGetWebHookDataMethodInfo()
+        {
+            return typeof(WebHookHandlerContextExtensions).GetMethod("GetDataOrDefault", BindingFlags.Public | BindingFlags.Static);
+        }
+
+        private static async Task<Dictionary<string, object>> GetFunctionArgumentsAsync(FunctionDescriptor function, HttpRequestMessage request)
+        {
+            ParameterDescriptor triggerParameter = function.Parameters.First(p => p.IsTrigger);
+            Dictionary<string, object> arguments = new Dictionary<string, object>();
+            object triggerArgument = null;
+            if (triggerParameter.Type == typeof(HttpRequestMessage))
+            {
+                triggerArgument = request;
+            }
+            else
+            {
+                // We'll replace the trigger argument but still want to flow the request
+                // so add it to the arguments, as a system argument
+                arguments.Add(ScriptConstants.DefaultSystemTriggerParameterName, request);
+
+                HttpTriggerBindingMetadata httpFunctionMetadata = (HttpTriggerBindingMetadata)function.Metadata.InputBindings.FirstOrDefault(p => p.Type == BindingType.HttpTrigger);
+                if (!string.IsNullOrEmpty(httpFunctionMetadata.WebHookType))
+                {
+                    WebHookHandlerContext webHookContext;
+                    if (request.Properties.TryGetValue(ScriptConstants.AzureFunctionsWebHookContextKey, out webHookContext))
+                    {
+                        triggerArgument = GetWebHookData(triggerParameter.Type, webHookContext);
+                    }
+                }
+
+                if (triggerArgument == null)
+                {
+                    triggerArgument = await request.Content.ReadAsAsync(triggerParameter.Type);
+                }
+            }
+
+            arguments.Add(triggerParameter.Name, triggerArgument);
```

Comment:
This is definitely something I was thinking about. I was planning on having the Node invoker take advantage of what is being added here when I work on the changes for #236.